store/mongo/datastore_mongo.go

```diff
@@ -810,50 +758,60 @@ func (db *DataStoreMongo) GetDevCountByStatus(ctx context.Context, status string
 		"$count": "count",
```

Comment:
Would you mind changing the nested `$cond` operators on line 674 (above) with a single `$switch`?
The amount of nesting there is already quite nauseating to look at :P